### PR TITLE
[TGUI] Telecomms

### PR DESCRIPTION
**Pull request by AffectedArc07**
## What Does This PR Do
This PR converts both tcomms machines to TGUI. There isnt really much more to say here, they both function the exact same.

## Why It's Good For The Game
Purge NanoUI

## Images of changes
**Core**
image

image

image

*Ion banner shows up on all pages*
image

**Relay**
image

image

## Changelog
:cl: AffectedArc07
add: Tcomms machines now use TGUI
/:cl:

### CODE REVIEW

File: code/game/machinery/tcomms/core.dm [hunk 1]
```diff
@@ -164,101 +164,90 @@
 // UI STUFF //
 //////////////
 
-/obj/machinery/tcomms/core/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
-	// this is silly but it has to be done because NTTC inits before languages do
+/obj/machinery/tcomms/core/tgui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/tgui_state/state = GLOB.tgui_default_state)
+	// This needs to happen here because of how late the language datum initializes. I dont like it
 	if(nttc.valid_languages.len == 1)
```
Comment (dearmochi) [inline code/game/machinery/tcomms/core.dm:169]: nit: While you're touching the proc, `if(length(nttc.valid_languages) == 1)`

File: tgui/packages/tgui/interfaces/TcommsCore.js [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+  );
+};
+
+const IonBanner = () => {
+  // This entire thing renders on one line
+  // Its just split in here to get past
+  // the 80 char line limit
+  return (
+    <NoticeBox>
+      ERROR: An Ionspheric overload has occured.
```
Comment (dearmochi) [inline tgui/packages/tgui/interfaces/TcommsCore.js:64]: `Ionspheric` -> `Ionospheric`

**Review by dearmochi — APPROVED** [review]
LGTM, even if I have no clue how Telecomms work

File: tgui/packages/tgui/interfaces/TcommsCore.js (hunk 2 above)
Comment (AffectedArc07) [reply tgui/packages/tgui/interfaces/TcommsCore.js:64]: Note to self: Update comment in `_base.dm`

File: code/game/machinery/tcomms/_base.dm [hunk 3]
```diff
@@ -40,7 +40,7 @@ GLOBAL_LIST_EMPTY(tcomms_machines)
 	var/network_id = "None"
 	/// Is the machine active
 	var/active = TRUE
-	/// Has the machine been hit by an ionspheric anomalie
+	/// Has the machine been hit by an ionospheric anomaly
```
Comment (AffectedArc07) [inline code/game/machinery/tcomms/_base.dm:43]: Edited because I cant spell

**Review by variableundefined — APPROVED** [review]

**Pull request merged**